src/auto-lightbox.js

```diff
@@ -25,14 +27,44 @@ export const AutoLightboxEvents = {
   NEWLY_SET: 'amp-auto-lightbox:newly-set',
 };
 
+/**
+ * @param {!./service/ampdoc-impl.AmpDoc} ampdoc
+ * @return {boolean}
+ */
+function isProxyOriginOrLocalDev(ampdoc) {
+  // Allow `localDev` in lieu of proxy origin for manual testing, except in
+  // tests where we need to actually perform the check.
+  const {win} = ampdoc;
+  if (getMode(win).localDev && !getMode(win).test) {
+    return true;
+  }
+
+  // An attached node is required for proxy origin check. If no elements are
+  // present, short-circuit.
+  const {firstElementChild} = ampdoc.getBody();
```

Comment:
Probably need to wait for body to be available first now that this is called on startup. Or you can check `isSingleDoc()` and pass `document.documentElement`.